### PR TITLE
feat: Add subscriptions executor command

### PR DESCRIPTION
**Pull request by lynnagara**
Subscription execution forms the second part of the subscriptions pipeline.
The executor's job is to consume the scheduled subscriptions from the scheduled
subscription topic, run the queries on ClickHouse and produce the results to
the results topic for Sentry.

This change simply introduces the executor CLI command. It builds the consumer
and processing strategy, however that strategy doesn't really do anything yet
except commit offsets. The processing strategies that are responsible for actually
running the query on ClickHouse and producing the results will be added separately.

Currently only a single dataset/entity pair is supported by the subscriptions
executor and a separate executor needs to be run for each entity.

### CODE REVIEW

File: snuba/subscriptions/executor_consumer.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+        assert (
+            storage is not None
+        ), f"Entity {entity_name} does not have a writable storage by default."
+
+        stream_loader = storage.get_table_writer().get_stream_loader()
+
+        scheduled_topic_spec = stream_loader.get_subscription_scheduled_topic_spec()
+        assert scheduled_topic_spec is not None
+        self.__scheduled_topic_spec = scheduled_topic_spec
```
Comment (fpacifici) [inline snuba/subscriptions/executor_consumer.py:58]: I believe the topic would be hardcoded here. Which is fine now. In the future we will likely want to make it possible to override via settings to be able to merge executor clusters or split executor clusters depending on how we want to deploy snuba.

**Review by fpacifici — CHANGES_REQUESTED** [review]
A couple of stylistic changes. Nothing major

File: snuba/cli/subscriptions_executor.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    "--dataset",
+    "dataset_name",
+    required=True,
+    type=click.Choice(["events", "transactions"]),
+    help="The dataset to target.",
+)
+@click.option(
+    "--entity",
+    "entity_name",
+    required=True,
```
[the remark this answers is not in the document]
Comment (lynnagara) [reply snuba/cli/subscriptions_executor.py:25]: I omitted this for now to keep the implementation simple (there are some other prerequisites for that), but we should support this

File: snuba/cli/subscriptions_executor.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    type=click.Choice(["events", "transactions"]),
+    help="The entity to target.",
+)
+@click.option(
+    "--consumer-group",
+    default="snuba-subscription-scheduler",
```
[the remark this answers is not in the document]
Comment (lynnagara) [reply snuba/cli/subscriptions_executor.py:31]: Yes, thanks

**Review by fpacifici — APPROVED** [review]

**Pull request merged**